installation/README.md

```diff
@@ -0,0 +1,161 @@
+# Production deployment using docker
+
+### Setup Letsencrypt Nginx Proxy Companion
+
+DNS needs to be configured for following to work
+
+Read more: https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion
+
+```sh
+cd $HOME
+git clone https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion.git
+cd docker-compose-letsencrypt-nginx-proxy-companion
+cp .env.sample .env
+./start.sh
+```
+
+### Clone frappe_docker repository
+
+```sh
+cd $HOME
+git clone https://github.com/frappe/frappe_docker.git
+cd frappe_docker
+cp installation/env-example .env
+
+# make directory for sites
+mkdir installation/sites
+```
+
+### Setup Environment Variables
+
+Environment variables are set to pass secret and variable data.
+
+If `env-example` is copied to `.env` following values are set.
+
+- `VERSION=edge` set version tag or latest for major version e.g. v12.3.0, v12
+- `MYSQL_ROOT_PASSWORD=admin`, set mariadb root password (bootstraps a mariadb container with this root password). If managed database mariadb is used NO need to set the password here
+- `MARIADB_HOST=mariadb` set hostname to `mariadb` in case of docker container for mariadb is used. In case managed db is used set the hostname/IP/domain name here
+- `SITES=site1.domain.com,site2.domain.com` these are list of sites that are part of the deployment "bench". Each site is separated by (,) comma
+- `LETSENCRYPT_EMAIL=your.email@your.domain.com` email for letsencrypt expiry notification
+
+### Start frappe-bench services
+
+```sh
+docker-compose \
+    --project-name frappebench00 \
+    -f installation/docker-compose-common.yml \
+    -f installation/docker-compose-erpnext.yml \
+    -f installation/docker-compose-networks.yml \
+    --project-directory installation up -d
+```
+
+Note: use `docker-compose-frappe.yml` in case you need bench with just frappe installed
+
+### Create new sites
+
+Note: Wait for mariadb to start. If new site creation fails re-try again after mariadb container is up and running
+
+```sh
+# Create ERPNext site
+docker exec -it \
+    -e "SITE_NAME=site1.domain.com" \
+    -e "DB_ROOT_USER=root" \
+    -e "DB_ROOT_PASSWORD=admin" \
+    -e "ADMIN_PASSWORD=admin" \
+    -e "INSTALL_ERPNEXT=1" \
+    frappebench00_erpnext-python_1 docker-entrypoint.sh new
```

Comment:
I wouldn't consider it the best practice to pass passwords on the command line. Instead I'd propose to pass `.env` file with these variables to `erpnext-pyton` container as `env_files` in `docker-compose-erpnext.yml`.

Reply:
Also instead of `DB_ROOT_PASSWORD` maybe let's use standard `MYSQL_ROOT_PASSWORD`, so that it's not confusing for end user.

Reply:
> I wouldn't consider it the best practice to pass passwords on the command line. Instead I'd propose to pass `.env` file with these variables to `erpnext-pyton` container as `env_files` in `docker-compose-erpnext.yml`.

I'd suggest create `.env` where needed or place them manually. Mentioned this in README

This docker-compose is just the place to check if things work in containers as basic single machine docker based production installation. In case of swarm or kubernetes, `secrets` can be used. Let this get merged. You can improve further or even add docker-compose for swarm